src/main/java/org/folio/bulkops/service/QueryService.java

```diff
@@ -1,94 +1,318 @@
+import com.opencsv.exceptions.CsvRequiredFieldEmptyException;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.log4j.Log4j2;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.folio.bulkops.client.QueryClient;
 import org.folio.bulkops.client.RemoteFileSystemClient;
+import org.folio.bulkops.client.SrsClient;
+import org.folio.bulkops.domain.bean.BulkOperationsEntity;
+import org.folio.bulkops.domain.bean.ExtendedHoldingsRecord;
+import org.folio.bulkops.domain.bean.ExtendedInstance;
+import org.folio.bulkops.domain.bean.ExtendedItem;
+import org.folio.bulkops.domain.bean.HoldingsRecord;
+import org.folio.bulkops.domain.bean.Instance;
+import org.folio.bulkops.domain.bean.Item;
+import org.folio.bulkops.domain.bean.StateType;
+import org.folio.bulkops.domain.bean.User;
+import org.folio.bulkops.domain.converter.JsonToMarcConverter;
 import org.folio.bulkops.domain.entity.BulkOperation;
+import org.folio.bulkops.domain.entity.BulkOperationExecutionContent;
+import org.folio.bulkops.exception.UploadFromQueryException;
+import org.folio.bulkops.processor.permissions.check.PermissionsValidator;
 import org.folio.bulkops.repository.BulkOperationRepository;
+import org.folio.bulkops.util.BulkOperationsEntityCsvWriter;
+import org.folio.bulkops.util.CSVHelper;
+import org.folio.querytool.domain.dto.QueryDetails;
 import org.folio.querytool.domain.dto.SubmitQuery;
+import org.folio.spring.FolioExecutionContext;
 import org.springframework.stereotype.Service;
 
-import java.io.ByteArrayInputStream;
-import java.time.LocalDateTime;
-import java.util.UUID;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.stream.Collectors;
-
 @Service
 @Log4j2
 @RequiredArgsConstructor
 public class QueryService {
   public static final String QUERY_FILENAME_TEMPLATE = "%1$s/Query-%1$s.csv";
 
-  private final QueryClient queryClient;
   private final BulkOperationRepository bulkOperationRepository;
-  private final RemoteFileSystemClient remoteFileSystemClient;
   private final ErrorService errorService;
 
+  private final FolioExecutionContext folioExecutionContext;
+
+  private final ObjectMapper objectMapper;
+
+  private final JsonToMarcConverter jsonToMarcConverter;
+
+  private final PermissionsValidator permissionsValidator;
+
+  private final RemoteFileSystemClient remoteFileSystemClient;
+  private final SrsClient srsClient;
+  private final QueryClient queryClient;
+
   private final ExecutorService executor = Executors.newCachedThreadPool();
 
-  public UUID executeQuery(SubmitQuery submitQuery) {
+  public BulkOperation retrieveRecordsAndCheckQueryExecutionStatus(BulkOperation bulkOperation) {
+    executor.execute(getRunnableWithCurrentFolioContext(() -> {
+      var queryId = executeQuery(new SubmitQuery(bulkOperation.getFqlQuery(), bulkOperation.getEntityTypeId()));
+      var queryResult = getResult(queryId);
```

Comment:
queryResult is used solely for monitoring the status of the FQM job at this point.
Therefore, there’s no need to fetch the full content - using /query/{{queryId}}?includeResults=false is sufficient at this stage to understand status of FQM job and avoid extra-traffic transferring.